core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2245,6 +2245,7 @@ object SparkContext extends Logging {
       if (activeContext.get() == null) {
         setActiveContext(new SparkContext(config), allowMultipleContexts = false)
       }
+      logWarning("Use an existing SparkContext, some configuration may not take effect.")
```

Comment:
maybe log this only if there are non-empty configs? same thing for all the warning messages later.